rust-src/wallet_library/src/identity.rs

```diff
@@ -0,0 +1,356 @@
+use anyhow::{bail, ensure, Result};
+use concordium_base::{
+    common::*,
+    id::{
+        account_holder::{generate_id_recovery_request, generate_pio_v1},
+        constants,
+        constants::ArCurve,
+        dodis_yampolskiy_prf as prf,
+        pedersen_commitment::Value as PedersenValue,
+        secret_sharing::Threshold,
+        types::*,
+    },
+};
+use key_derivation::{ConcordiumHdWallet, Net};
+use serde::{Deserialize as SerdeDeserialize, Serialize as SerdeSerialize};
+use serde_json::{json, to_string};
+use std::{collections::BTreeMap, convert::TryInto};
+
+use crate::wallet::get_wallet;
+
+type JsonString = String;
+
+/// Defines the JSON structure that matches the output of the function
+/// generating the identity objcect request.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+struct IdentityObjectRequestV1 {
+    id_object_request: Versioned<PreIdentityObjectV1<constants::IpPairing, ArCurve>>,
+}
+
+/// The shared identity creation request input that is independent
+/// of whether or not the request is going to be created
+/// by supplying a seed phrase or the secret key material
+/// directly.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdRequestCommonInput {
+    ip_info:        IpInfo<constants::IpPairing>,
+    global_context: GlobalContext<constants::ArCurve>,
+    ars_infos:      BTreeMap<ArIdentity, ArInfo<constants::ArCurve>>,
+    net:            Net,
```

Comment:
The `net` should be moved from the common input to the `seed` specific input. It is not used when providing the keys directly, only when deriving keys from the seed.